superset/views/core.py

```diff
@@ -1237,9 +1238,12 @@ def explore_json(self, datasource_type=None, datasource_id=None):
         requests that GETs or POSTs a form_data.
 
         `self.generate_json` receives this input and returns different
-        payloads based on the request args in the first block
+        payloads based on the request args in the first block"""
 
-        TODO: break into one endpoint for each return shape"""
+        if request.method == 'GET' and not is_feature_enabled('CLIENT_CACHE'):
+            raise MethodNotAllowed('POST')
```

Comment:
```suggestion
            raise MethodNotAllowed(request.method)
```
The method that's not allowed is the method that was used, right? `POST` is allowed.

Reply:
According to docs: https://werkzeug.palletsprojects.com/en/0.15.x/exceptions/, first parameter is valid_methods. Added argument name here.